spark_pipeline_framework/transformers/http_data_sender/v1/http_data_sender_processor.py

```diff
@@ -40,32 +113,34 @@ def send_partition_to_server(
         if len(json_data_list) == 0:
             yield Row(url=None, status=0, result=None, request_type=None, headers=None)
 
+        headers["Content-Type"] = content_type
+        if oauth_enabled:
+            access_token = HttpDataSenderProcessor.create_access_token(
+                client_id, auth_url, client_secret
+            )
+            headers.update({"Authorization": f"Bearer {access_token}"})
+
         assert url
         json_data: Dict[str, Any]
         for json_data in json_data_list:
-            headers["Content-Type"] = content_type
-            request: HelixHttpRequest = HelixHttpRequest(
-                request_type=RequestType.POST,
+            create_request = partial(
+                HttpDataSenderProcessor.create_request,
                 url=url,
-                headers=headers,
-                payload=json_data,
                 post_as_json_formatted_string=post_as_json_formatted_string,
+                json_data=json_data,
+                parse_response_as_json=parse_response_as_json,
             )
-            if parse_response_as_json:
-                response_json = request.get_result()
-                yield Row(
-                    url=url,
-                    status=response_json.status,
-                    result=response_json.result,
-                    headers=json.dumps(headers, default=str),
-                    request_type=str(RequestType.POST),
-                )
-            else:
-                response_text = request.get_text()
-                yield Row(
-                    url=url,
-                    status=response_text.status,
-                    result=response_text.result,
-                    headers=json.dumps(headers, default=str),
-                    request_type=str(RequestType.POST),
-                )
+            row: Optional[Row] = None
+            try:
+                row = create_request(headers=headers)
+            except requests.exceptions.HTTPError as e:
+                if oauth_enabled and getattr(e.response, "status", None) == 401:
+                    access_token = HttpDataSenderProcessor.create_access_token(
+                        client_id, auth_url, client_secret
+                    )
+                    headers.update({"Authorization": f"Bearer {access_token}"})
+                    row = create_request(headers=headers)
+                else:
+                    raise e
```

Comment:
check how to add multiple retries logic. We should be able to retry for example 3 times